packages/jazz-tools/src/implementation/zodSchema/zodCo.ts

```diff
@@ -26,68 +27,49 @@ import {
 import { RichTextSchema } from "./schemaTypes/RichTextSchema.js";
 import { z } from "./zodReExport.js";
 
+function enrichCoMapSchema<Shape extends z.core.$ZodLooseShape>(
+  schema: z.ZodObject<
+    { -readonly [P in keyof Shape]: Shape[P] },
+    z.core.$strip
+  >,
+) {
+  const baseCatchall = schema.catchall;
+
+  const enrichedSchema = Object.assign(schema, {
+    collaborative: true,
+    create: (...args: any[]) => {
+      return coSchema.create(...args);
+    },
+    load: (...args: any[]) => {
+      return coSchema.load(...args);
+    },
+    subscribe: (...args: any[]) => {
+      return coSchema.subscribe(...args);
+    },
+    findUnique: (...args: any[]) => {
+      return coSchema.findUnique(...args);
+    },
+    catchall: (index: z.core.$ZodType) => {
+      return enrichCoMapSchema(baseCatchall(index));
+    },
+    withHelpers: (helpers: (Self: z.core.$ZodType) => object) => {
+      return Object.assign(schema, helpers(schema));
```

Comment:
The object spread breaks both the recursive props and misses some not-enumerable props required for `z.discriminatedUnion` to work.

Refactored `catchall` to not use the spread while keeping the schema immutability.
Moved to `Object.assign` for withHelpers.